packages/wayland_proxy_virtwl.rb

```diff
@@ -0,0 +1,143 @@
+           let wayland_clipboard_offer = ref None in
+      -    let primary_device = Primary_mgr.get_device primary_selection_mgr ~seat @@ primary_device ~wayland_primary_offer in
+           let clipboard_device = Clipboard_mgr.get_data_device clipboard_mgr ~seat @@ clipboard_device ~wayland_clipboard_offer in
+           {
+             x11;
+      @@ -495,8 +474,6 @@ module Selection = struct
+             wayland_primary_offer;
+             selection_window;
+             set_selection_window;
+      -      primary_selection_mgr;
+      -      primary_device;
+             (* Clipboard: *)
+             wayland_clipboard_offer;
+             clipboard_window;
+    PATCH_XWAYLAND_EOF
+    @relay_patch = <<~'PATCH_RELAY_EOF'
+      --- a/wayland-proxy-virtwl/relay.ml	2021-11-18 20:34:27.000000000 +0000
+      +++ b/wayland-proxy-virtwl/relay.ml	2021-11-24 23:48:42.805651473 +0000
+      @@ -466,13 +466,6 @@ let make_surface ~xwayland ~host_surface
+             if x#scale <> 1l then
+               H.Wl_surface.set_buffer_scale h ~scale:x#scale;       (* Xwayland will be a new enough version *)
+             let set_configured s =
+      -        if s = `Unmanaged then (
+      -          (* For pointer cursors we want them at the normal size, even if low-res.
+      -             Also, Vim tries to hide the pointer by setting a 1x1 cursor, which confuses things
+      -             when unscaled. Ideally we would stop doing transforms in this case, but it doesn't
+      -             seem to matter. *)
+      -          H.Wl_surface.set_buffer_scale h ~scale:1l;
+      -        );
+               state := s;
+               match data.state with
+               | Ready | Destroyed -> ()
+    PATCH_RELAY_EOF
+    File.write('xwayland.patch', @xwayland_patch)
+    system 'patch -Np2 -i xwayland.patch'
+    File.write('relay.patch', @relay_patch)
+    system 'patch -Np2 -i relay.patch'
+  end
+
+  def self.install
+    ENV['OPAMROOT'] = @OPAMROOT
+    FileUtils.mkdir_p "#{CREW_DEST_PREFIX}/bin"
+    if (ARCH == 'armv7l') || (ARCH == 'aarch64')
+      # ld.gold breaks logs 0.7.0 builds with ocaml 4.1.2
+      FileUtils.cp "#{CREW_PREFIX}/bin/ld.bfd", "#{CREW_DEST_PREFIX}/bin/ld"
+    end
+    system "PATH=#{CREW_DEST_PREFIX}/bin:$PATH OPAMROOT=#{@OPAMROOT} opam install . --root=#{@OPAMROOT} --destdir=#{CREW_DEST_DIR}#{@OPAMROOT} -y"
+    FileUtils.rm "#{CREW_DEST_PREFIX}/bin/ld" if File.exist?("#{CREW_DEST_PREFIX}/bin/ld")
+    Dir.chdir "#{CREW_DEST_PREFIX}/bin" do
+      FileUtils.ln_s '../share/opam/bin/wayland-proxy-virtwl', 'wayland-proxy-virtwl'
+    end
+  end
+
+  def self.postinstall
+    puts <<~'EOFSTRING'.lightblue
+      Note that there is not yet hardware acceleration in wayland-proxy-virtwl.
+      wayland-proxy-virtwl example usage (before running a gui program):
+    EOFSTRING
+    puts <<~'EOFSHELLCODE'.lightcyan
+      wayland-proxy-virtwl --wayland-display wayland-2 --x-display=2 --xrdb Xft.dpi:150 &
```

Comment:
Get the dpi value using xdpyinfo like sommelier does?
https://github.com/skycocker/chromebrew/blob/cb9a768917622f73ea0513010fb5ac4e8fd1009c/packages/sommelier.rb#L179

Reply:
That would be a good idea for a shell script for automatic setup, for sure.

(Also that's a chicken and egg issue... `xdpyinfo` won't work until a X11 Display is setup no?)

Reply:
I think the point of this PR is to offer a proof of concept... not to immediately replace `sommelier`.

Reply:
> That would be a good idea for a shell script for automatic setup, for sure.
> 
> (Also that's a chicken and egg issue... `xdpyinfo` won't work until a X11 Display is setup no?)

You’re right😅 Never mind